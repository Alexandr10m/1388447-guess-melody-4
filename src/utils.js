const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const noop = () => {};

export {extend, noop};
