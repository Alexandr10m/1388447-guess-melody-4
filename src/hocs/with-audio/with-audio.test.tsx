import * as React from "react";
import * as renderer from "react-test-renderer";
import withAudio from "./with-audio";
import {noop} from "../../utils";


interface MockComponent {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponent) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withAudio(MockComponent);

describe(`Snapshot of withAudio`, () => {
  it(`withAudio is rendered correctly`, () => {
    const tree = renderer
      .create((
        <MockComponentWrapped
          isPlaying={false}
          onPlayButtonClick={noop}
          src={``}
        />
      ), {
        createNodeMock() {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
