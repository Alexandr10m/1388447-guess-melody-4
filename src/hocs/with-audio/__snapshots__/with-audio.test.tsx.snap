// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of withAudio withAudio is rendered correctly 1`] = `
<div>
  <audio />
</div>
`;
